src/test/java/hudson/plugins/xvnc/XvncTest.java

```diff
@@ -134,6 +147,118 @@ public void avoidNpeAfterDeserialiation() throws Exception {
         j.buildAndAssertSuccess(p);
     }
 
+    public void testXauthorityInWorkspace() throws Exception {
+        DumbSlave slave = j.createOnlineSlave();
+
+        FreeStyleProject job = j.jenkins.createProject(FreeStyleProject.class, "jobA");
+        job.setAssignedNode(slave);
+
+        runXvnc(job, true);
+
+        job.getBuildersList().add(new TestBuilder() {
+            @Override
+            public boolean perform(AbstractBuild<?, ?> build, Launcher launcher, BuildListener listener) throws InterruptedException, IOException {
+                FilePath[] list = build.getWorkspace().list(".Xauthority-*");
+                assertEquals("There should be one Xauthority file in the workspace", 1, list.length);
```

Comment:
If the workspace doesn't have a space in the path then it will be created in the workspace, doesn't matter how the other paths look, the workspace is the priority.